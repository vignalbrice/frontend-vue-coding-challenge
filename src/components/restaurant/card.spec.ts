import { mount } from '@vue/test-utils';

import RestaurantCard from './card.vue';
import restaurantsResponse from '@/mock/restaurants.json';

const restaurant = restaurantsResponse[0];

describe(`restaurant-card`, () => {
  test(`no props`, (context) => {
    const wrapper = mount(RestaurantCard, {
      global: { plugins: [context.router] },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it(`renders correctly a restaurant`, (context) => {
    const wrapper = mount(RestaurantCard, {
      global: { plugins: [context.router] },
      props: { restaurant },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
