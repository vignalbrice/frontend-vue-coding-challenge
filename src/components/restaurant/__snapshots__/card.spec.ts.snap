// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`restaurant-card > no props 1`] = `
"<div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\">
  <!---->
  <div class=\\"v-card__loader\\">
    <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
      <!---->
      <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
      <div class=\\"v-progress-linear__indeterminate\\">
        <div class=\\"v-progress-linear__indeterminate long\\"></div>
        <div class=\\"v-progress-linear__indeterminate short\\"></div>
      </div>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <v-img-stub height=\\"100\\" cover=\\"true\\" eager=\\"false\\" options=\\"[object Object]\\" src=\\"\\" inline=\\"false\\" transition=\\"fade-transition\\"></v-img-stub>
  <div class=\\"v-card-title\\"></div>
  <div class=\\"v-rating v-rating--hover v-theme--light\\"><label for=\\"v-rating-0-0\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 0 of 5</span>
      <!---->
    </label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-0\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"0\\">
    <div class=\\"v-rating__wrapper\\">
      <!---->
      <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-1\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 1 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--variant-plain\\" style=\\"height: 32px; width: 32px;\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
            <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star-outline mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
            <!---->
            <!---->
          </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-1\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"1\\"></div>
      <!---->
    </div>
    <div class=\\"v-rating__wrapper\\">
      <!---->
      <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-2\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 2 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--variant-plain\\" style=\\"height: 32px; width: 32px;\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
            <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star-outline mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
            <!---->
            <!---->
          </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-2\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"2\\"></div>
      <!---->
    </div>
    <div class=\\"v-rating__wrapper\\">
      <!---->
      <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-3\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 3 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--variant-plain\\" style=\\"height: 32px; width: 32px;\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
            <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star-outline mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
            <!---->
            <!---->
          </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-3\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"3\\"></div>
      <!---->
    </div>
    <div class=\\"v-rating__wrapper\\">
      <!---->
      <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-4\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 4 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--variant-plain\\" style=\\"height: 32px; width: 32px;\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
            <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star-outline mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
            <!---->
            <!---->
          </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-4\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"4\\"></div>
      <!---->
    </div>
    <div class=\\"v-rating__wrapper\\">
      <!---->
      <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-5\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 5 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--variant-plain\\" style=\\"height: 32px; width: 32px;\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
            <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star-outline mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
            <!---->
            <!---->
          </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-5\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"5\\"></div>
      <!---->
    </div>
  </div>
  <div class=\\"v-card-text\\">
    <div class=\\"flex gap-2\\"><i class=\\"mdi-map-marker mdi v-icon notranslate v-theme--light v-icon--size-default text-slate-300\\" aria-hidden=\\"true\\"></i>
      <div class=\\"text-slate-600\\"><pre class=\\"my-0\\"></pre>
      </div>
    </div>
  </div>
  <div class=\\"v-card-actions mt-auto\\"><a class=\\"v-btn v-theme--light text-primary v-btn--density-default v-btn--size-default v-btn--variant-tonal\\" href=\\"/restaurants/undefined\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
      <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"> see more </span>
      <!---->
      <!---->
    </a></div>
  <!---->
  <!----><span class=\\"v-card__underlay\\"></span>
</div>"
`;

exports[`restaurant-card > renders correctly a restaurant 1`] = `
"<div class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\">
  <!---->
  <div class=\\"v-card__loader\\">
    <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
      <!---->
      <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
      <div class=\\"v-progress-linear__indeterminate\\">
        <div class=\\"v-progress-linear__indeterminate long\\"></div>
        <div class=\\"v-progress-linear__indeterminate short\\"></div>
      </div>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <v-img-stub height=\\"100\\" cover=\\"true\\" eager=\\"false\\" options=\\"[object Object]\\" src=\\"https://s3-media2.fl.yelpcdn.com/bphoto/RDKmsxZZczP4EiTCB9AYOw/o.jpg\\" inline=\\"false\\" transition=\\"fade-transition\\"></v-img-stub>
  <div class=\\"v-card-title\\">Billy The Butcher</div>
  <div class=\\"v-rating v-rating--hover v-theme--light\\"><label for=\\"v-rating-0-0\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 0 of 5</span>
      <!---->
    </label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-0\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"0\\">
    <div class=\\"v-rating__wrapper\\">
      <!---->
      <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-1\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 1 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--variant-plain\\" style=\\"color: #ffd055; caret-color: #ffd055; height: 32px; width: 32px;\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
            <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
            <!---->
            <!---->
          </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-1\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"1\\"></div>
      <!---->
    </div>
    <div class=\\"v-rating__wrapper\\">
      <!---->
      <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-2\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 2 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--variant-plain\\" style=\\"color: #ffd055; caret-color: #ffd055; height: 32px; width: 32px;\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
            <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
            <!---->
            <!---->
          </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-2\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"2\\"></div>
      <!---->
    </div>
    <div class=\\"v-rating__wrapper\\">
      <!---->
      <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-3\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 3 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--variant-plain\\" style=\\"color: #ffd055; caret-color: #ffd055; height: 32px; width: 32px;\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
            <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
            <!---->
            <!---->
          </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-3\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"3\\"></div>
      <!---->
    </div>
    <div class=\\"v-rating__wrapper\\">
      <!---->
      <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-4\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 4 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--variant-plain\\" style=\\"color: #ffd055; caret-color: #ffd055; height: 32px; width: 32px;\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
            <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
            <!---->
            <!---->
          </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-4\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"4\\"></div>
      <!---->
    </div>
    <div class=\\"v-rating__wrapper\\">
      <!---->
      <div class=\\"v-rating__item\\"><label for=\\"v-rating-0-5\\" class=\\"\\"><span class=\\"v-rating__hidden\\">Rating 5 of 5</span><button type=\\"button\\" class=\\"v-btn v-btn--icon v-theme--light v-btn--density-default v-btn--variant-plain\\" style=\\"height: 32px; width: 32px;\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
            <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><i class=\\"mdi-star-outline mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\"></i></span>
            <!---->
            <!---->
          </button></label><input class=\\"v-rating__hidden\\" name=\\"v-rating-0\\" id=\\"v-rating-0-5\\" type=\\"radio\\" tabindex=\\"-1\\" value=\\"5\\"></div>
      <!---->
    </div>
  </div>
  <div class=\\"v-card-text\\">
    <div class=\\"flex gap-2\\"><i class=\\"mdi-map-marker mdi v-icon notranslate v-theme--light v-icon--size-default text-slate-300\\" aria-hidden=\\"true\\"></i>
      <div class=\\"text-slate-600\\"><pre class=\\"my-0\\">Jungfernstieg 16 - 20
4. OG
20354 Hamburg
Germany</pre>
      </div>
    </div>
  </div>
  <div class=\\"v-card-actions mt-auto\\"><a class=\\"v-btn v-theme--light text-primary v-btn--density-default v-btn--size-default v-btn--variant-tonal\\" href=\\"/restaurants/7974729\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
      <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"> see more </span>
      <!---->
      <!---->
    </a></div>
  <!---->
  <!----><span class=\\"v-card__underlay\\"></span>
</div>"
`;
